src/TensorValues/Operations.jl

```diff
@@ -207,18 +207,18 @@ end
   Meta.parse("ThirdOrderTensorValue{$D}($str)")
 end
 
-# a_il = b_ij*c_jl
-@generated function dot(a::A,b::B) where {A<:MultiValue{Tuple{D1,D2}},B<:MultiValue{Tuple{D2,D3}}} where {D1,D2,D3}
-  ss = String[]
-  for l in 1:D3
-    for i in 1:D1
-      s = join([ "a[$i,$j]*b[$j,$l]+" for j in 1:D2])
-      push!(ss,s[1:(end-1)]*", ")
-    end
-  end
-  str = join(ss)
-  Meta.parse("TensorValue{$D1}($str)")
-end
+# a_il = b_ij*c_jl # Duplicate of L157
+# @generated function dot(a::A,b::B) where {A<:MultiValue{Tuple{D1,D2}},B<:MultiValue{Tuple{D2,D3}}} where {D1,D2,D3}
+#   ss = String[]
+#   for l in 1:D3
+#     for i in 1:D1
+#       s = join([ "a[$i,$j]*b[$j,$l]+" for j in 1:D2])
+#       push!(ss,s[1:(end-1)]*", ")
+#     end
+#   end
+#   str = join(ss)
+#   Meta.parse("TensorValue{$D1}($str)")
+# end
 
```

Comment:
The commented method seems wrong. It is supposed to be a single contraction of a 2-tensors but you return a 1-tensor. It should return a 2-tensor, which is the method implemented in L157. I would eliminate this commented part.

Reply:
I think that because of the number of inputs, it defaults to a 2-tensor. Regardless, the code can be removed. Will do.